src/main/java/org/openlmis/stockmanagement/web/ValidReasonAssignmentController.java

```diff
@@ -153,7 +153,9 @@ private void checkIsValidRequest(ValidReasonAssignment assignment) {
         .findByProgramIdAndFacilityTypeIdAndReasonId(programId, facilityTypeId, reasonId);
 
     if (foundAssignment != null) {
-      return new ResponseEntity<>(reasonAssignmentDtoBuilder.build(foundAssignment), OK);
+      throw new ValidationMessageException(
```

Comment:
After the refactor, the method name is missleading. It will no longer "find existing or save". It will either save a new assignemnt or fail to do so - we need to update the method name so that the code remains clean